src/bundle.rs

```diff
@@ -14,66 +14,134 @@ pub const BUNDLE_ROOTFS: &str = "rootfs";
-/// and return the final bundle config path.
+/// Convert an `application/vnd.oci.image.config.v1+json` object into an OCI runtime configuration
+/// blob and write to `config.json`.
+///
+/// [OCI Image Spec: Conversion to OCI Runtime Configuration](https://github.com/opencontainers/image-spec/blob/main/conversion.md)
+/// states:
+///
+/// The "default generated runtime configuration" MAY be overridden or combined with externally
+/// provided inputs from the caller. In addition, a converter MAY have its own
+/// implementation-defined defaults and extensions which MAY be combined with the "default generated
+/// runtime configuration".
 pub fn create_runtime_config(
     image_config: &ImageConfiguration,
     bundle_path: &Path,
 ) -> Result<PathBuf> {
     let mut spec = Spec::default();
+    let mut annotations: HashMap<String, String> = HashMap::new();
+    let mut labels = HashMap::new();
 
+    // Update the default hostname of `youki`
     spec.set_hostname(Some(BUNDLE_HOSTNAME.to_string()));
-    let mut annotations: HashMap<String, String> = HashMap::new();
-    annotations.insert(ANNOTATION_OS.to_string(), image_config.os().to_string());
-    annotations.insert(
-        ANNOTATION_ARCH.to_string(),
-        image_config.architecture().to_string(),
-    );
-    if let Some(author) = image_config.author() {
-        annotations.insert(ANNOTATION_AUTHOR.to_string(), author.to_string());
-    }
-    if let Some(created) = image_config.created() {
-        annotations.insert(ANNOTATION_CREATED.to_string(), created.to_string());
-    }
-    let mut process = Process::default();
 
     if let Some(config) = image_config.config() {
+        let mut process = Process::default();
+
+        // Verbatim Fields:
+        //
+        // A compliant configuration converter MUST extract the following fields verbatim to the
+        // corresponding field in the generated runtime configuration:
+        // - WorkingDir
+        // - Env
+        // - EntryPoint
+        // - Cmd
         if let Some(working_dir) = config.working_dir() {
             process.set_cwd(PathBuf::from(working_dir));
         }
-
+        // The converter MAY add additional entries to process.env but it SHOULD NOT add entries
+        // that have variable names present in Config.Env.
         if let Some(env) = config.env() {
             process.set_env(Some(env.to_vec()));
         }
-
+        // If both Config.Entrypoint and Config.Cmd are specified, the converter MUST append the
+        // value of Config.Cmd to the value of Config.Entrypoint and set process.args to that
+        // combined value.
         let mut args: Vec<String> = vec![];
         if let Some(entrypoint) = config.entrypoint() {
             args.extend(entrypoint.clone());
         }
-
         if let Some(cmd) = config.cmd() {
             args.extend(cmd.clone());
         }
-
         if !args.is_empty() {
             process.set_args(Some(args));
         }
 
```

Comment:
There have a bug in original code, could help fix it together with this PR?
we need update the `process` in `spec` which is missing in original code :-(
```
spec.set_process(Some(process));
```

Reply:
Yes, I also found this issue, but missed it.